felix/dataplane/linux/int_dataplane.go

```diff
@@ -807,10 +809,33 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 	dp.RegisterManager(newFloatingIPManager(natTableV4, ruleRenderer, 4, config.FloatingIPsEnabled))
 	dp.RegisterManager(newMasqManager(ipSetsV4, natTableV4, ruleRenderer, config.MaxIPSetSize, 4))
 	if config.RulesConfig.IPIPEnabled {
+		var routeTableIPIP routetable.RouteTableInterface
+		if !config.RouteSyncDisabled {
+			log.Debug("RouteSyncDisabled is false.")
+			ipipEncapProto := defaultRoutingProto
+			if config.DeviceRouteProtocol != syscall.RTPROT_BOOT {
+				ipipEncapProto = config.DeviceRouteProtocol
+			}
+			routeTableIPIP = routetable.New([]string{"^" + IPIPIfaceName + "$"}, 4, config.NetlinkTimeout,
+				config.DeviceRouteSourceAddress, ipipEncapProto, true, unix.RT_TABLE_MAIN,
+				dp.loopSummarizer, featureDetector, routetable.WithLivenessCB(dp.reportHealth))
+		} else {
+			log.Info("RouteSyncDisabled is true, using DummyTable.")
+			routeTableIPIP = &routetable.DummyTable{}
+		}
 		log.Info("IPIP enabled, starting thread to keep tunnel configuration in sync.")
 		// Add a manager to keep the all-hosts IP set up to date.
-		dp.ipipManager = newIPIPManager(ipSetsV4, config.MaxIPSetSize, config.ExternalNodesCidrs)
-		go dp.ipipManager.KeepIPIPDeviceInSync(config.IPIPMTU, config.RulesConfig.IPIPTunnelAddress, dataplaneFeatures.ChecksumOffloadBroken)
+		dp.ipipManager = newIPIPManager(
+			ipSetsV4,
+			routeTableIPIP,
+			IPIPIfaceName,
+			config,
+			dp.loopSummarizer,
+			4,
+			featureDetector,
+		)
+		dp.ipipParentC = make(chan string, 1)
```

Comment:
Does IPIP have the concept of a parent device?  I thought that was a quirk of VXLAN.  I'd expect that aspect not to be needed.  Of course, if we do the IPIP device set-up in the background then we still need a kick channel but I don't think you need to send the parent device?

Reply:
AFAIK, the parent device is needed to add routes to no encap routes. And for IPIP `CrossSubnet` we need to add no encap routes, right?